plugins/cpp/parser/CMakeLists.txt

```diff
@@ -1,5 +1,16 @@
+# Set LLVM version to 11.1 if using Ubuntu 22
+execute_process(COMMAND bash -c get_ubuntu_version.sh
+        OUTPUT_VARIABLE UBUNTU_VERSION
+        RESULT_VARIABLE VERSION_RESULT
+        OUTPUT_STRIP_TRAILING_WHITESPACE)
+if (${VERSION_RESULT} EQUAL 0)
+    if (${UBUNTU_VERSION} GREATER_EQUAL 22)
+        message(STATUS "LLVM version override to 11.1 because of Ubuntu 22")
+        set(LLVM_VERSION_TO_FIND 11.1)
+    endif ()
+endif ()
 # Find LLVM
-find_package(LLVM REQUIRED CONFIG)
+find_package(LLVM ${LLVM_VERSION_TO_FIND} REQUIRED CONFIG)
```

Comment:
This does not seem really robust. A proper solution would be simply the following:
```cmake
find_package(LLVM 11 REQUIRED CONFIG)
```

However, semantic versioning is not handled correctly, so e.g. version 11.1 does not meet the requirement  to match version 11. Since CMake 3.19, there are version ranges, so we can write this:
```cmake
find_package(LLVM 11...<12 REQUIRED CONFIG)
```

But on Ubuntu 20.04, we only have CMake 3.16 from the official package repository. So until we drop support for Ubuntu 20.04, you shall either find a more robust solution, which does not strictly ties the version requirement to `11.1`, as other minor versions are fine as well, or leave this issue (out of scope for the ticket), and the proper LLVM and Clang paths can be given to CMake with the `LLVM_DIR` and `Clang_DIR` flags, as the [build guide](https://github.com/Ericsson/CodeCompass/blob/master/doc/deps.md#build-codecompass) describes.